Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -524,44 +529,44 @@ void HTMLInputElement::updateType(const AtomString& typeAttributeValue)
     m_hasType = true;
     if (!newType)
         return;
-    ASSERT(m_inputType->type() != newType->type());
+    ASSERT(protectedInputType()->type() != newType->type());
```

Comment:
We should avoid repeated calls to `protectedInputType()` in this function. I know m_inputType gets updated mid-way through this function but we should still cache it in a local variable to avoid ref counting churn.